03-securing-communication/README.md

```diff
@@ -0,0 +1,480 @@
+* it doesn't maintain an active connection
+* it's a request-response protocol
+* it provides a series of codes to mark the type of request and replies
+* HTTP requests consists of paths (routes) that are mapped to resources
+
+The lack of an active connection session is compensated by the use of cookies.
+Similarly, the plain text nature of the protocol means that anyone could read contents in HTTP network packets.
+This is alleviated by the use of HTTPS.
+Nowadays, most connections use HTTPS precisely because of the need for confidentiality.
+
+In the previous session, you used web browsers and the [Developer Tools](https://developer.mozilla.org/en-US/docs/Learn/Common_questions/What_are_browser_developer_tools) feature of modern browsers to inspect traffic, update cookies, inspect rendered pages.
+GUI web browsers (such as Firefox, Chrome, Edge, Safari) provide the appealing interface for users to surf the web.
+For quick and dirty tasks such as testing connections, making requests and downloading large files, we use CLI web clients such as `curl` and `wget`.
+We will be using these in this session as well.
+
+## Confidentiality
+
+Confidentiality is a security property that prevents captured data from being understood by an attacker.
+If an attacker captures data with confidentiality ensured, the attacher must not be able to extract actual information from it.
+Confidentiality is generally provided with encryption.
+
+For example, a classical HTTP connection is plain text and thus non-confidential.
+Let's exemplify this.
+On one terminal, start a `tcpdump` capture session for HTTP connections:
+
+```
+$ sudo tcpdump -A tcp port 80
+```
+
+On another terminal, make an HTTP connection using `curl`:
+
+```
+$ curl http://elf.cs.pub.ro
+```
+
+As the connection is HTTP, you will see plain text messages as part of the `tcpdump` output:
+
+```
+Host: elf.cs.pub.ro
+User-Agent: curl/7.58.0
+Accept: */*
+
+[...]
+Date: Sun, 03 Jul 2022 15:51:46 GMT
+Server: Apache/2.4.38 (Debian)
+Last-Modified: Mon, 02 Aug 2010 17:58:06 GMT
+ETag: "a8-48cdaf14da780"
+Accept-Ranges: bytes
+Content-Length: 168
+Vary: Accept-Encoding
+Content-Type: text/html
+
+<html>
+	<head>
+		<meta name="google-site-verification" content="gTsIxyV43HSJraRPl6X1A5jzGFgQ3N__hKAcuL2QsO8" />
+	</head>
+
+	<body>
+		<h1>It works!</h1>
+	</body>
+</html>
+```
+
+However, if we used `curl` and an HTTPS connection:
+
+```
+$ curl https://elf.cs.pub.ro
+```
+
+there would be no plain text output, because the connection is using a confidential (encrypted) channel.
+
+The same traffic inspection can be done in a more visual manner using Wireshark.
+
+As long as traffic is not encrypted, an attacker able to capture packets (either fooling someone to get the data or simply accessing a networking equipment along the way) will read the traffic contents.
+HTTPS uses public key cryptography to ensure the confidentiality of network traffic.
+
+## Public Key Cryptography. Certificates
```

Comment:
I think we should add something about how private/public keys works on certificate signing because it might be confusing for that situation to see that encryption is made with private key.